### ISSUE TITLE
Suggestion: Move to another unit test framework

### ISSUE THREAD

**Issue by TsudaKageyu** [taglib/taglib#822]
**CppUnit** haven't been able to catch up the latest compilers. I found that it doesn't work well with the latest AppleClang without an option and I made a [quick workaround](https://github.com/taglib/taglib/commit/6000a19f70ef57f993b7f5594a2dcf9651e5801d). 
Additionally, I have thought all along that CppUnit is not very handy. For example, it requires useless type conversions like:
```
    // Why I can't compare TagLib::String to a string literal?
    CPPUNIT_ASSERT_EQUAL(f.tag()->title(), String("Title1"));

    // I'd like to simply do:
    CPPUNIT_ASSERT_EQUAL(f.tag()->title(), "Title1");
```

So I'd like to suggest that we move from CppUnit to another up-to-date unit test framework.
I privately tried [Boost.Test](http://www.boost.org/doc/libs/1_64_0/libs/test/doc/html/index.html) and it worked fine. (The diff is [here](https://github.com/TsudaKageyu/taglib/commit/8e544d13443210115f8e2727b757b66e7c02c229).) Other libraries like [Google Test](https://github.com/google/googletest) or [Catch](https://github.com/philsquared/Catch) would be great.

**Comment by sbooth** [comment]
I agree, I'm not a fan of CppUnit.  I always like boost but I don't know enough about Google Test or Catch to propose a selection.

**Comment by TsudaKageyu** [comment]
I have tried another library [Catch](https://github.com/philsquared/Catch). (See 41968bb192a671c040c7be5bff40afcd5a0c8dd2) I find it really handy and love it.
Especially, it doesn't require any preparation (like building static libs, adding paths and blah blah blah) even used with Visual Studio since it's a single-header library. Of course, it works fine on either Linux or OSX too.

I'd like to create a pull request based on the commit 41968bb192a671c040c7be5bff40afcd5a0c8dd2.
